### DISCUSSION TITLE
drainage and invasion percolation input parameters

### DISCUSSION THREAD

**Discussion by JahanzebTariq** [PMEAL/OpenPNM#2673] [Q&A]
greetings. I was running the drainage and invasion simulation but got stuck in the parameters putting process, as the pores in my coding are named ans 'xmin' and 'xmax' instead of left and right pores, but when I enter 'xmin' and 'xmax', I get error.
![image](https://user-images.githubusercontent.com/99289270/214768638-8e1c5f5d-7407-453e-9706-d92bd74392e2.png)
![image](https://user-images.githubusercontent.com/99289270/214768762-1534fdfa-f5f4-4694-9fe9-c80b2b181ef0.png)

but this does not occur if I use left and right despite of the fact that I have been using 'xmin' and 'xmax' everywhere in the code.
![image](https://user-images.githubusercontent.com/99289270/214768407-2f59bf76-512d-4a79-90b7-feab5782150d.png)

**please address this issue**
thank you

**Comment by jgostick** [comment]
You can see what labels are present on the network by doing `print(pn)`.  Snow returns `xmin`, `xmax` etc, but generators, like Cubic, return `left`, `right`, etc.  Maybe we would unify these...I will actually create an issue that we should add `xmin` etc to the generators as well, though keeping `left` `right` too.
